### PR TITLE
Spooky Room Info - Cyclic Reference Avoidance

### PR DESCRIPTION
**Pull request by BajaTheFrog**
Proposing a possible way to organize code that gets around the cyclic reference issue

### CODE REVIEW

File: project/player_character/player_character.gd [hunk 1]
```diff
@@ -15,7 +15,7 @@ onready var summoning_power := $SummoningPower as GhostSummonerComponent
 onready var door_exit = Vector2.ZERO
 onready var can_travel = false
 onready var available_action = PlayerActions.NONE setget action_setter
-onready var current_room
+onready var current_room_info = SpookyRoomInfo
```
Comment (jcoates) [inline project/player_character/player_character.gd:18]: I guess this is currently unused? But I wonder if the player will want the position, or if they need a handle on the kind of room they're in.

File: project/spooky_room/spooky_room.gd [hunk 2]
```diff
@@ -64,10 +64,10 @@ func _process(delta):
 func _on_RoomArea2D_body_entered(body):
 	if body.is_in_group(Constants.GROUP_PLAYER):
 		var player = body as PlayerCharacter
-		player.current_room = self
+		player.current_room_info = SpookyRoomInfo.new(self.global_position)
```
Comment (jcoates) [inline project/spooky_room/spooky_room.gd]: ah is global position like, baked in?

File: project/player_character/player_character.gd (hunk 1 above)
Comment (jcoates) [reply project/player_character/player_character.gd:18]: Knowing the room you're in might be useful for like, checking if you're in the room with a ghost if that's how that works. But we could do that with this I guess, as I continue thinking. With like an enum for room name or something,

**Review by NathanDHolmes — APPROVED** [review]

**Comment by NathanDHolmes** [comment]
lgtm

**Pull request merged**